twine/repository.py

```diff
@@ -91,12 +98,21 @@ def close(self) -> None:
 
     @staticmethod
     def _convert_data_to_list_of_tuples(data: Dict[str, Any]) -> List[Tuple[str, Any]]:
+        # This does what ``warehouse.forklift.parse_form_metadata()`` does, in reverse.
         data_to_send = []
         for key, value in data.items():
-            if key in KEYWORDS_TO_NOT_FLATTEN or not isinstance(value, (list, tuple)):
-                data_to_send.append((key, value))
-            else:
+            key = KEYWORDS_TO_RENAME.get(key, key)
+            if isinstance(value, (list, tuple)) and key not in KEYWORDS_TO_NOT_FLATTEN:
                 data_to_send.extend((key, item) for item in value)
+            elif key == "project_urls":
+                assert isinstance(value, dict), "for mypy"
+                for name, url in value:
+                    data_to_send.append((key, f"{name}, {url}"))
```

Comment:
I'm confused here. Is `value` a `dict` with keys that are `(name, url)` pairs?

I removed this elif block, and the tests still pass. If this code is necessary, it needs some test coverage.

Reply:
`twine.package.PackageMetadata` is a typed dictionary, if you look at its type annotation you will see that indeed `project_urls` is a dictionary of `name` and `url` pairs. I see that the code contains a bug, thus tests are needed, there are no existing tests for this function.

Reply:
I added a test.